### PR TITLE
Enable govt cloud

### PR DESCRIPTION
**Pull request by aaomidi**
I haven't tested the govt cloud integration itself. Will need to sync up on how to test for national/govt clouds.

### CODE REVIEW

File: src/sql/workbench/contrib/accounts/browser/accountDialog.ts [hunk 1]
```diff
@@ -192,8 +196,29 @@ export class AccountDialog extends Modal {
...
+				});
+
+				const picked = await this._quickInputService.pick(buttons, { canPickMany: false });
+
+				pickedValue = picked?.label;
+			} else {
+				pickedValue = vals[0].view.title;
+			}
+
+			const v = vals.filter(v => v.view.title === pickedValue)?.[0];
```
Comment (anthonydresser) [inline src/sql/workbench/contrib/accounts/browser/accountDialog.ts]: If these values are undefined, shouldn't you show something to the user?

File: src/sql/workbench/contrib/accounts/browser/accountDialog.ts [hunk 2]
```diff
@@ -192,8 +196,29 @@ export class AccountDialog extends Modal {
...
+
+			v?.addAccountAction?.run();
```
Comment (anthonydresser) [inline src/sql/workbench/contrib/accounts/browser/accountDialog.ts]: same here.

File: src/sql/workbench/contrib/accounts/browser/accountDialog.ts (hunk 1 above)
Comment (aaomidi) [reply src/sql/workbench/contrib/accounts/browser/accountDialog.ts]: Yeah I think that'd be a good idea.

File: extensions/azurecore/package.json [hunk 3]
```diff
@@ -41,6 +41,39 @@
             "description": "%config.enablePublicCloudDescription%"
           }
         }
+      },
+      {
+        "type": "object",
+        "title": "Azure Account Configuration",
```
Comment (abist) [inline extensions/azurecore/package.json]: You might want to have a localized title which is pulled from the `package.nls.json` for the new properties

Comment (aaomidi) [reply extensions/azurecore/package.json]: How would I reference that from here?

Comment (abist) [reply extensions/azurecore/package.json]: Something similar to https://github.com/microsoft/azuredatastudio/pull/8984/files#diff-a7e46294cf1dea22425a9e834e76e256R25

File: extensions/azurecore/package.json [hunk 4]
```diff
@@ -41,6 +41,39 @@
             "description": "%config.enablePublicCloudDescription%"
           }
         }
+      },
+      {
+        "type": "object",
```
Comment (Charles-Gagnon) [inline extensions/azurecore/package.json]: You can just combine these all into the existing Azure Account Configuration (just add new property objects). See https://github.com/microsoft/azuredatastudio/blob/master/extensions/machine-learning-services/package.json#L32 for an example

File: src/sql/workbench/contrib/accounts/browser/accountDialog.ts [hunk 5]
```diff
@@ -192,8 +196,34 @@ export class AccountDialog extends Modal {
...
+			}
+
+			const v = vals.filter(v => v.view.title === pickedValue)?.[0];
+
+			if (!v) {
+				this._notificationService.error(localize('accountDialog.noCloudsRegistered', "You have no clouds enabled. Go to Settings -> Search Azure Account Configuration -> Enable at least one cloud"));
+				return;
+			}
+
+			v.addAccountAction.run();
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/accounts/browser/accountDialog.ts]: Is this a dangling promise here?

**Comment by aaomidi** [comment]
This is blocked by ADAL until further notice.

**Pull request closed without merging**